spec/unit/lib/wifi_user/use_cases/check_if_whitelisted_email_spec.rb

```diff
@@ -0,0 +1,35 @@
+describe WifiUser::UseCases::CheckIfWhitelistedEmail do
+  subject { described_class.new(gateway: s3_gateway) }
```

Comment:
I'd call this `spy_gateway` or similar in a spec.

Reply:
Not really a spy though, will call it regex_gateway_stub

Reply:
If you are cool w that